File: src/multiLanguageCodes/WriteTime.go
package main

import (
	"fmt"
	"log"
	"os"
	"time"
)

// Main function to run the logging program.
func main() {
	const filename = "MYFILE.txt"

	// Create a log message with a timestamp.
	nowtime := time.Now().Format("2006/01/02 15:04:05")
	logMessage := fmt.Sprintf("This program is written in Go.\nCurrent Time = %s\n", nowtime)

	// Attempt to write the log message to the file.
	if err := writeLog(filename, logMessage); err != nil {
		log.Fatalf("Program terminated with an error: %v", err)
	}

	fmt.Printf("Successfully appended log to %s.\n", filename)
}

// writeLog appends a given message to a file, creating it if it doesn't exist.
func writeLog(filename string, message string) error {
	// Open the file in append mode. Create it if it doesn't exist.
	// Use read-write permissions 0644 (read/write for owner, read-only for others).
	file, err := os.OpenFile(filename, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
	if err != nil {
		return fmt.Errorf("failed to open file: %w", err)
	}
	defer file.Close()

	// Write the message to the file.
	if _, err := fmt.Fprint(file, message); err != nil {
		return fmt.Errorf("failed to write to file: %w", err)
	}

	return nil
}

File: src/multiLanguageCodes/write_time.c
#include <stdio.h>
#include <time.h>
#include <string.h>
#include <stdbool.h>

#define FILENAME "MYFILE.txt"
#define TIMESTAMP_BUFFER_SIZE 20

/**
 * @brief Appends a timestamped log message to a file.
 *
 * @param filename The path to the file.
 * @return Returns true on success, false on failure.
 */
bool writeTime(const char* filename) {
    FILE* outfile = NULL;
    time_t t;
    struct tm* tm_info;
    char nowtime[TIMESTAMP_BUFFER_SIZE];

    // Open file in append mode.
    outfile = fopen(filename, "a");
    if (outfile == NULL) {
        perror("Failed to open file");
        return false;
    }

    // Get the current system date and time.
    t = time(NULL);
    tm_info = localtime(&t);
    if (tm_info == NULL) {
        fprintf(stderr, "Failed to get local time.\n");
        fclose(outfile);
        return false;
    }

    // Format the timestamp string.
    strftime(nowtime, TIMESTAMP_BUFFER_SIZE, "%Y/%m/%d %H:%M:%S", tm_info);

    // Write the messages to the file.
    fprintf(outfile, "This program is written in C.\n");
    fprintf(outfile, "Current Time = %s\n\n", nowtime);

    // Close the file.
    fclose(outfile);
    return true;
}

int main() {
    // Attempt to write the log and check the return value.
    if (writeTime(FILENAME)) {
        printf("Successfully appended log to %s.\n", FILENAME);
    } else {
        // Error message is already printed inside writeLog.
        printf("Program terminated with an error.\n");
        return 1;
    }

    return 0;
}
